amazon-kinesis-client/src/test/java/software/amazon/kinesis/utils/StreamExistenceManager.java

```diff
@@ -73,12 +98,101 @@ public List<String> getAllResourceNames() throws Exception {
+                putResourcePolicyForCrossAccount(streamArn,
+                        getCrossAccountStreamResourcePolicy(testConfig.getAccountIdForConsumer(), streamArn));
+            }
+        }
+    }
+
+    public Map<Arn, Arn> createCrossAccountConsumerIfNecessary() throws Exception {
+        // For cross account, KCL cannot create the consumer automatically in another account, so
+        // we have to create it ourselves and provide the arn to the StreamConfig in multi-stream mode or
+        // RetrievalConfig in single-stream mode
+        if (testConfig.isCrossAccount() && testConfig.getRetrievalMode().equals(RetrievalMode.STREAMING)) {
+            final Map<Arn, Arn> streamToConsumerArnsMap = new HashMap<>();
+            for (Arn streamArn : testConfig.getStreamArns()) {
+                final Arn consumerArn = registerConsumerAndWaitForActive(streamArn,
+                        KCLAppConfig.CROSS_ACCOUNT_CONSUMER_NAME);
+                putResourcePolicyForCrossAccount(consumerArn,
+                        getCrossAccountConsumerResourcePolicy(testConfig.getAccountIdForConsumer(), consumerArn));
+                streamToConsumerArnsMap.put(streamArn, consumerArn);
+            }
+            return streamToConsumerArnsMap;
+        }
+        return null;
+    }
+
+    private void putResourcePolicyForCrossAccount(Arn resourceArn, String policy) {
+        try {
+            final PutResourcePolicyRequest putResourcePolicyRequest = PutResourcePolicyRequest.builder()
+                    .resourceARN(resourceArn.toString())
+                    .policy(policy)
+                    .build();
+            FutureUtils.resolveOrCancelFuture(client.putResourcePolicy(putResourcePolicyRequest), Duration.ofSeconds(60));
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to PutResourcePolicy "  + policy + " on resource " + resourceArn, e);
+        }
+    }
+
+    private String getCrossAccountStreamResourcePolicy(String accountId, Arn streamArn) {
+        return "{\"Version\":\"2012-10-17\","
+                + "\"Statement\":[{"
+                + "\"Effect\": \"Allow\","
+                + "\"Principal\": {\"AWS\": \"" + accountId + "\"},"
+                + "\"Action\": ["
+                + "\"kinesis:DescribeStreamSummary\",\"kinesis:ListShards\",\"kinesis:PutRecord\",\"kinesis:PutRecords\","
+                + "\"kinesis:GetRecords\",\"kinesis:GetShardIterator\"],"
+                + "\"Resource\": \"" + streamArn.toString() + "\""
+                + "}]}";
+    }
+
+    private String getCrossAccountConsumerResourcePolicy(String accountId, Arn consumerArn) {
+        return "{\"Version\":\"2012-10-17\","
+                + "\"Statement\":[{"
+                + "\"Effect\": \"Allow\","
+                + "\"Principal\": {\"AWS\": \"" + accountId + "\"},"
+                + "\"Action\": ["
+                + "\"kinesis:DescribeStreamConsumer\",\"kinesis:SubscribeToShard\"],"
```

Comment:
nit: can optionally extract the remainder of this string into common code to reduce duplication